.github/workflows/codeql.yml

```diff
@@ -17,14 +35,11 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        branch: [ 'master' ] # release branches are scanned in teleport-sec-scan repos (see RFD 147)
         language: [ 'go', 'javascript' ]
 
     steps:
     - name: Checkout repository
       uses: actions/checkout@v4
-      with:
-        ref: ${{ matrix.branch }}
```

Comment:
This was preventing the default checkout behavior, resulting in codeql getting very confused as to why it was on master, when it was expecting a PR ref.

Since we no longer run on other branches in this repo, we no longer need a matrix.